File: solutions/7.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
 public:
  int reverse(int x) {
    int temp = abs(x);
    long long ans = 0;
    while (temp > 0) {
      ans = ans * 10 + temp % 10;
      temp /= 10;
    }
    if (x > 0)
      return ans > INT_MAX || ans < INT_MIN ? 0 : ans;
    else
      return ans > INT_MAX || ans < INT_MIN ? 0 : -(ans);
  }
};

File: solutions/7.py
class Solution:
    def reverse(self, x: int) -> int:
        y=1
        if x<0:
            x=abs(x)
            y=-1
        s=list(str(x))[::-1]
        x=int("".join(s))
        x=x*y
        if x < -2**31 or x > 2**31 - 1:
            return 0
        return x 
